implants/lib/eldritch/src/random/string_impl.rs

```diff
@@ -0,0 +1,49 @@
+use anyhow::Result;
+use rand::{distributions::Alphanumeric, Rng}; // 0.8
+
+pub fn string(length: u64) -> Result<String> {
+    let s: String = rand::thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(length as usize)
+        .map(char::from)
+        .collect();
+    return Ok(s)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    const NUM_ITERATION: i32 = 1000000;
+
+    #[test]
+    fn test_string() -> anyhow::Result<()> {
+        let rand_string = string(5)?;
+        assert_eq!(rand_string.chars().count(), 5);
+        Ok(())
+    }
+
+    #[test]
+    fn test_string_uniform() -> anyhow::Result<()> {
+        use std::collections::HashSet;
+
+        let mut result_str = HashSet::new();
+        for _ in 0..=NUM_ITERATION {
+            let new_str = string(16)?;
+            assert_eq!(new_str.chars().count(), 16);
+            if !result_str.insert(new_str){
+                assert!(false);
+            }
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn test_string_length() -> anyhow::Result<()> {
+        for i in 0..=1000 {
```

Comment:
I like it!
Maybe instead we can do 0, 1000, u64::max to really put it through its paces without adding too many iterations.

Reply:
u64::max is apparently too big (exceeds max vector size). Added a warning about large random strings to the docs and set the test to `[0, 1000, 8192000]`